src/data_base.cpp

```diff
@@ -324,6 +324,31 @@ code data_base::reorganize(const config::checkpoint& fork_point,
     return result ? error::success : error::operation_failed;
 }
 
+code data_base::confirm(const hash_digest& block_hash,
```

Comment:
This approach is simple but least optimal. By populating a set and then using the transactions.confirm() overload that walks the set, there is an unnecessary memory bump (up to ~4mb for BTC with segwit) during confirmation. This and the additional iteration can be avoided by confirming in the population loop.

Much more importantly, the read of the entire tx is defeating the intended optimization. The objective here is to read the set of tx offsets and then confirm each transaction without reading the entire block. This requires confirming using `database::transaction_result` while avoiding transition through `chain::transaction`. This allows the tx to be confirmed without being fully populated. transaction_result is not fully-deferred, so this is also not entirely optimal, but is at least the correct abstraction.

For much greater optimization the tx confirmation loop can be parallelized by tx, but let's investigate that as a later optimization once this is solid.